james/errors.go

```diff
@@ -16,30 +15,33 @@ type JamesServerError struct {
 	Details    string `json:"details"`
 }
 
+func (e JamesServerError) Error() string {
+	return fmt.Sprintf("james server responded with status code: %d, type: %s, message: %s, details: %s\n", e.StatusCode, e.Type, e.Message, e.Details)
+}
+
 func unmarshalToJamesServerError(r *http.Response) *JamesServerError {
-	bodyData, err := io.ReadAll(r.Body)
-	if err != nil {
+	if r == nil {
```

Comment:
Get err and r also. And check if r is nil if nil then we make the struct from the err otherwise we use err.

Reply:
done